### PR TITLE
Plugins: Update backend plugin debugging instructions

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**
The `-debug` flag is no longer necessary for debugging backend plugins, so the documentation should be updated.

**Why do we need this feature?**

Docs would be stale otherwise

**Who is this feature for?**

Plugin developers

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/developers/plugins/development-with-local-grafana.md [hunk 1]
```diff
@@ -174,5 +174,5 @@ Configure your code editor to run the following steps:
 ### Notes
 
 - All logs are printed in the plugin's `stdout` rather than in Grafana logs.
-- If the backend plugin doesn't serve requests after you turn off debug mode, you can force a reset to the standalone mode. To do so, delete the files `dist/standalone.txt`, `dist/pid.txt`, and the executable file, and then restart Grafana.
+- If the backend plugin doesn't serve requests after you stop debugging, you can force a reset to the standalone mode. To do so, delete the files `dist/standalone.txt`, `dist/pid.txt`, and the executable file, and then restart Grafana.
```
Comment (wbrowne) [inline docs/sources/developers/plugins/development-with-local-grafana.md:177]: Do you think this line still makes sense even @xnyo?

File: docs/sources/developers/plugins/development-with-local-grafana.md [hunk 2]
```diff
@@ -157,13 +157,13 @@ If you re-run the configuration, Grafana automatically reloads the plugin.
 
 Configure your code editor to run the following steps:
 
-1. Build the executable file with debug flags.
+1. Build the executable file.
```
[the remark this answers is not in the document]
Comment (wbrowne) [reply docs/sources/developers/plugins/development-with-local-grafana.md:160]: Ah my bad. I thought this was just doing more `-debug` stuff!

**Review by xnyo — APPROVED** [review]
LGTM! Do we need to backport this so it shows also in the main docs, or no?

File: docs/sources/developers/plugins/development-with-local-grafana.md [hunk 3]
```diff
@@ -157,13 +157,13 @@ If you re-run the configuration, Grafana automatically reloads the plugin.
...
    ```
-   mage build:debug
+   mage build
```
[the remark this answers is not in the document]
Comment (wbrowne) [reply docs/sources/developers/plugins/development-with-local-grafana.md]: Fixed!

**Comment by wbrowne** [comment]
> LGTM! Do we need to backport this so it shows also in the main docs, or no?

Hmm.. This can be a bit confusing since the docs are bundled with Grafana, but the actual change is now released in the SDK. Were you thinking it makes sense to roll it into the next patch rather than `10.1.0`?

**Comment by xnyo** [comment]
> Were you thinking it makes sense to roll it into the next patch rather than 10.1.0?

Yeah pretty much. However if the -debug flag is present it shouldn't cause any issues and should behave the same.

So yeah let's not backport it and keep it like this :+1:

**Review by josmperez — APPROVED** [review]
LGTM from a style perspective.

**Review by marefr — APPROVED** [review]

**Review by mckn — APPROVED** [review]
LGTM!!

**Pull request merged**